File: pat1079.py
"""
    思路：1.用二维列表存储节点数据，如[[1,2,3],[],[]],列表为空代表没有孩子节点
          2.用字典存储某叶节点出售商品数量
          3.用深度优先遍历遍历整棵树

          注：python输入函数太耗时了，导致2，3，5，6测试点运行超时

"""



origin = input().split()
N = int(origin[0])
price = float(origin[1])
rate = float(origin[2])
tree = []
amount_dic = {}         #   记录零售商出售商品数量的字典
for i in range(N):
    children_input = [int(x) for x in input().split()]
    if children_input[0] != 0:
        tree.append(children_input[1:])
    else:
        tree.append([])
        amount_dic[i] = children_input[1]

total_sales = 0.0
def dfs(root, depth):
    global total_sales      #   没有这句话会导致编译错误
    if len(tree[root]) == 0:
        total_sales += (price * 
                        (1 + rate * 0.01) ** depth * amount_dic[root])
    else:
        for child in tree[root]:
            dfs(child,depth + 1)
    



dfs(0,0)
print("{:.1f}".format(total_sales))

File: pat1079.cpp
/*
	思路：通过深度遍历优先方法遍历树，在每个叶节点处计算价钱
		
	注：1.用cin输入会超时
		2.浮点数据类型用double，不要float，否则会出现计算错误
*/


#include<iostream>
#include<vector>
#include<cmath>
using namespace std;

struct node{
	int amount = 0;
	vector<int> children;

}Tree[100010];

double total_sales = 0.0;
double price, rate;
void dfs(int root, int depth) {
	int len = Tree[root].children.size();		//	孩子节点个数
	if (len == 0) {
		total_sales += price *
			pow((1 + rate * 0.01), depth ) * Tree[root].amount;
	}
	else {
		for (int i = 0; i < len; i++) {
			dfs(Tree[root].children[i], depth + 1);
		}
	}
}

int main() {

	int N;
	cin >> N >> price >> rate;
	for (int i = 0; i < N; i++) {
		int k;
		scanf("%d", &k);
		if (k != 0) {
			for (int j = 0; j < k; j++) {
				int child;
				scanf("%d", &child);
				Tree[i].children.push_back(child);
			}
		}
		else{
			int amount;
			scanf("%d", &amount);
			Tree[i].amount = amount;
		}
	}
	dfs(0, 0);
	printf("%.1f", total_sales);

	return 0;
}